packages/cozy-konnector-libs/src/libs/error.js

```diff
@@ -1,19 +1,55 @@
 const log = require('cozy-logger').namespace('Error Interception')
 
-// This will catch exception which would be uncaught by the connector script itself
-process.on('uncaughtException', err => {
+const handleUncaughtException = err => {
   log('critical', err.message, 'uncaught exception')
   process.exit(1)
-})
-process.on('unhandledRejection', err => {
+}
+
+const handleUnhandledRejection = err => {
   log('critical', err.message, 'unhandled exception')
   process.exit(1)
-})
-process.on('SIGTERM', () => {
+}
+
+const handleSigterm = () => {
   log('critical', 'The konnector got a SIGTERM')
   process.exit(128 + 15)
-})
-process.on('SIGINT', () => {
+}
+
+const handleSigint = () => {
   log('critical', 'The konnector got a SIGINT')
   process.exit(128 + 2)
-})
+}
+
+let attached = false
+
+/**
+ * Attach event handlers to catch uncaught exceptions/rejections and signals.
+ * Log them as critical and exit the process accordingly.
+ * If the cleanup function has not been called, calling again the function
+ * is a no-op.
+ *
+ * @param  {Process} prcs - Process object, default to current process
+ * @return {Function} When called, removes the signal handlers
+ */
+const attachProcessEventHandlers = (prcs = process) => {
+  if (attached) {
+    return
+  }
+  attached = true
+  prcs.on('uncaughtException', handleUncaughtException)
+  prcs.on('unhandledRejection', handleUnhandledRejection)
+  prcs.on('SIGTERM', handleSigterm)
+  prcs.on('SIGINT', handleSigint)
+
+  return () => {
+    attached = false
+    prcs.off('uncaughtException', handleUncaughtException)
```

Comment:
`EventEmitter.off` is marked as added in node v10 (https://nodejs.org/api/events.html#events_emitter_off_eventname_listener) and we still use node v8 in production.

Unit tests worked in node 8 anyway (no idea why) but could you use `removeListener` instead?

Reply:
Done, thanks :)